### ISSUE TITLE
Error in rollmean(): Parameter 'size' is too large.

### ISSUE THREAD

**Issue by privefl** [privefl/bigutilsr#2]
E.g. `bigutilsr::rollmean(x = 1:3, size = 2)`.
It expects that `(2 * size + 1) > length(x)`.

What strategy to adopt when it is not the case? (+ recode)

**Comment by shadrinams** [comment]
Any updates on the issue?

**Comment by privefl** [comment]
I still don't know what would be the best way to handle this, and whether I should.
Indeed, if the `size` of the smoothing is too large, you'll get the mean for all values, which is not very useful anyway.

Any better suggestion?

**Comment by alexey-larionov** [comment]
I came across this error when I tried bed_autoSVD() function with a very small number of variants (I know that normally several thousand of variants should be used; however, occasionally I have to analyse some awkward datasets).  

To deal with this error I just explicitly suppress the smoothing log-p-values (as it doesn't make sense in this context) by setting roll.size=0.  

On my opinion, it might be practically helpful, if you changed the default behaviour to something similar to what I described above.  E.g. change Error to Warning, so the function completes, but warns the user that the smoothing had been suppressed.

**Comment by privefl** [comment]
Yes, we could default back to 0, or to the largest allowed value, or just implement a version that works even if the size is too large.
I don't really have an opinion on which solution would be the best, this is why I haven't really done anything to solve this yet.
